### PR TITLE
libfive: update outdated version string

### PR DESCRIPTION
**Pull request by Wulfsta**
## Description of changes

Missed updating the version string in #288922, correct it here.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/libfive/default.nix [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 stdenv.mkDerivation {
   pname = "libfive";
-  version = "unstable-2023-06-07";
+  version = "unstable-2024-02-14";
```
Comment (kirillrdy) [inline pkgs/development/libraries/libfive/default.nix:20]:
```suggestion
  version = "0-unstable-2024-02-14";
```

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

Comment (Wulfsta) [reply pkgs/development/libraries/libfive/default.nix]: Resolved - didn't know about this.

**Review by kirillrdy — APPROVED** [review]

**Review by jopejoe1 — APPROVED** [review]
Maybe add an update script so that doesn't happen in the future?

```nix
passthru.updateScript = unstableGitUpdater { };
```

**Comment by Wulfsta** [comment]
> Maybe add an update script so that doesn't happen in the future?
> 
> ```nix
> passthru.updateScript = unstableGitUpdater { };
> ```

Done, tested locally and it works! That said, it doesn't currently seem to conform to the naming standard (it does not prefix a `0-`), but it's still nice.

**Comment by kirillrdy** [comment]
> > Maybe add an update script so that doesn't happen in the future?
> > ```nix
> > passthru.updateScript = unstableGitUpdater { };
> > ```
> 
> Done, tested locally and it works! That said, it doesn't currently seem to conform to the naming standard (it does not prefix a `0-`), but it's still nice.

yes it's a known issue

https://github.com/NixOS/nixpkgs/pull/280501

**Review by drupol — APPROVED** [review]

**Pull request merged**